### PR TITLE
User validation mapper

### PR DESCRIPTION
**Pull request by Bozhokmaria**

### CODE REVIEW

File: src/main/java/com/exadel/discountwebapp/user/mapper/UserMapper.java [hunk 1]
```diff
@@ -3,27 +3,19 @@
...
 public class UserMapper {
 
-    public static User toEntity(UserRequestVO requestVO){
-        return User.builder()
-                .firstName(requestVO.getFirstName())
-                .lastName(requestVO.getLastName())
-                .email(requestVO.getEmail())
-                .password(requestVO.getPassword())
-                .build();
+    private ModelMapper modelMapper = new ModelMapper();
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/user/mapper/UserMapper.java]: add Strict mapping strategy

Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/user/mapper/UserMapper.java]: done

File: src/main/java/com/exadel/discountwebapp/user/mapper/UserMapper.java [hunk 2]
```diff
@@ -3,27 +3,27 @@
...
-                .build();
+    private final ModelMapper modelMapper = new ModelMapper();
+
+    @PostConstruct
+    public void init(){
+        modelMapper.getConfiguration().setMatchingStrategy(MatchingStrategies.STRICT);
+    }
+
+    public User toEntity(UserRequestVO requestVO){
+        return modelMapper.map(requestVO, User.class);
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/user/mapper/UserMapper.java]: do we need toEntity() method?

File: src/main/java/com/exadel/discountwebapp/user/entity/User.java [hunk 3]
```diff
@@ -28,16 +29,20 @@
...
     private String lastName;
 
-    @Column(name = "usr_email", length = 255, unique = true, nullable = false)
+    @NotNull
+    @Column(name = "usr_email", length = 255, unique = true)
     private String email;
 
-    @Column(name = "usr_password", length = 510, nullable = false)
+    @NotNull
+    @Column(name = "usr_password", length = 510)
```
Comment (ahlukhouski2) [inline src/main/java/com/exadel/discountwebapp/user/entity/User.java:45]: location/role also not null?

File: src/main/java/com/exadel/discountwebapp/user/mapper/UserMapper.java (hunk 2 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/user/mapper/UserMapper.java]: due to requirements we don't have opportunity to create or update(for now) users in db, so this method is unnecessary. deleted

File: src/main/java/com/exadel/discountwebapp/user/entity/User.java (hunk 3 above)
Comment (Bozhokmaria) [reply src/main/java/com/exadel/discountwebapp/user/entity/User.java:45]: added

**Review by ahlukhouski2 — APPROVED** [review]

**Pull request merged**